gr-digital/lib/constellation.cc

```diff
@@ -239,16 +245,22 @@ void constellation::gen_soft_dec_lut(int precision, float npwr)
 {
     d_soft_dec_lut.clear();
     d_lut_scale = powf(2.0f, static_cast<float>(precision));
+    //check if noise power is overriden. use global if not
+    if (npwr != d_npwr && npwr != 1.0){d_npwr = npwr;}
+    else {npwr = d_npwr;}
 
     // We know we've normalized the constellation, so the min/max
     // dimensions in either direction are scaled to +/-1.
     float maxd = 1.0f;
+    //the above comment isn't really true, but LUT is funky so we'll scale our points inside the +-1 LUT
+    //we will multiply inputs to obtain LLR's for points with the scale ptscale instead of +-1
+    float ptscale = 2.0f*d_maxamp;
     float step = (2.0f * maxd) / (d_lut_scale - 1);
     float y = -maxd;
     while (y < maxd + step) {
         float x = -maxd;
         while (x < maxd + step) {
-            gr_complex pt = gr_complex(x, y);
+            gr_complex pt = gr_complex(x*ptscale, y*ptscale);
             d_soft_dec_lut.push_back(calc_soft_dec(pt, npwr));
             x += step;
         }
```

Comment:
I suggest rewriting this like so:
```c++
    float step = (2.0f * ptscale) / (d_lut_scale - 1);
    float y = -ptscale;
    while (y < ptscale + step) {
        float x = -ptscale;
        while (x < ptscale + step) {
            gr_complex pt = gr_complex(x, y);
            d_soft_dec_lut.push_back(calc_soft_dec(pt, npwr));
            x += step;
        }
 ```
 I think this empasizes the fact that we're sweeping a grid of `d_lut_scale` points covering the interval `[-ptscale, ptscale]`, with both `x` and `y`. Once we do this, we can remove the variable `maxd`, I think.

Reply:
I think this should be fine, I'll give it a try Monday and report/commit if all is good.

Reply:
@daniestevez Naively copy and pasting this seemed to result in the test FG hanging forever. Not sure what the problem is, but I do not have time to troubleshoot unfortunately.

Reply:
No worries. I don't understand why my suggestion doesn't work and your code does. I thought them to be equivalent (basically replacing `maxd by `ptscale`). After reading your comment https://github.com/gnuradio/gnuradio/pull/6478#discussion_r1089902370, I think that there might be some subtleties that don't make this replacement straightforward, so I'm happy to leave the code as it is currently, given that it works.